cmd/fever/cmds/run.go

```diff
@@ -189,6 +190,33 @@ func mainfunc(cmd *cobra.Command, args []string) {
 			}
 		}
 		forwardHandler.(*processing.ForwardHandler).Run()
+
+		stenosis := viper.GetBool("stenosis.enable")
+		if stenosis {
+			// flow notifier setup
+			flowNotifyChan := make(chan types.Entry)
+			defer close(flowNotifyChan)
+			fn := processing.MakeFlowNotifier(flowNotifyChan)
+			dispatcher.RegisterHandler(fn)
+			var tlsConfig *tls.Config
+			if viper.GetBool("stenosis.tls") {
+				tlsConfig, err = util.MakeTLSConfig(viper.GetString("stenosis.client-chain-file"),
+					viper.GetString("stenosis.client-key-file"),
+					viper.GetStringSlice("stenosis.root-cas"),
+					viper.GetBool("stenosis.skipverify"))
+			}
+			if err != nil {
```

Comment:
This if-err should be part of the previous if-block, otherwise the err from previous scope will be used.

```
if viper.GetBool("stenosis.tls") {
  var err error
  tlsConfig, err = ...
  if err != nil {
    ...
}
``

Reply:
Good catch, thanks!